spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/AbstractMessageBusBinderPlugin.java

```diff
@@ -132,10 +146,57 @@ protected final void bindConsumerAndProducers(Module module) {
 			if (isTapActive(tapChannelName)) {
 				createAndBindTapChannel(tapChannelName, outputChannel);
 			}
+			if (trackHistory) {
+				track(module, outputChannel, historyProperties);
+			}
 		}
 		MessageChannel inputChannel = module.getComponent(MODULE_INPUT_CHANNEL, MessageChannel.class);
 		if (inputChannel != null) {
 			bindMessageConsumer(inputChannel, getInputChannelName(module), properties[0]);
+			if (trackHistory && module.getType().equals(ModuleType.sink)) {
```

Comment:
We only want to track each module once; I chose to track the output channel but, of course, the last module (sink) doesn't have an output channel, so we track on its input channel instead.